### ISSUE TITLE
How do I define schema definitions inline?

### ISSUE THREAD

**Issue by StianOvrevage** [readmeio/swagger-inline#5]
How do I define schema definitions inline?

Having to define schema separately kind of defeats the whole purpose of inline-documentation. :)

**Comment by dwasyluk** [comment]
I like this in concept but wonder what a practical implementation would be. Having all the model definitions in swaggerBase makes sense from a maintenance and coordination perspective. 

It would be odd to have endpoints in multiple files using the same model schema but then having to remember where that schema was defined inline.

**Comment by gkoberger** [comment]
Does anyone have any suggestions? Right now, we just do it in the swagger base file. I'd be happy to include it if we could come up with some simple syntax that makes sense!

**Comment by gkoberger** [comment]
I'm not necessarily against this, but out of curiosity, why would you prefer this to having it in a YAML/JSON file?

I'm slightly worried about people having multiple same-named schema's in various places, and having them collide.

**Comment by glock18** [comment]
Disclaimer: I already mustered up a decent solution for my case allowing me to minimize effort of generating an openapi.json, but I'll share my thoughts on the topic.

I see your points. Here are my objections to why I think having this option is not a bad thing:

1. It supports the separation of concerns. With a modular structure it makes sense to have routes and models (and anything else really when it comes to openapi - like responses, tags) to be in more or less the same place. Centralization of, say, models would depreciate the separation and make the maintenance a bit awkward. 

2. With a big number of models, responses, and such, the base file is going to be huge, and it more or less eliminates the purpose of this package how I see it, that is having the openapi documentation stored along with the relevant code.

3. Most importantly it gives an option without taking an option of using a mega base file away.

However there are cons, that I concur to agree with:
1. In order to give this option, it needs to be implemented, thus there needs to be a volunteer this endeavour.
2. It doesn't respect the fact that models may already be described in some other way (like using Joi in my case), where it would be awkward to do the same job (i.e. description) twice for the same entity.

For now I used this package and it served my needs marvellously! Albeit it didn't magically built my entire openapi.json from scratch it did what it claimed to do and in tandem with other tools I managed to build a feasible solution to the problem. Thank you for this simple yet wonderful tool!
